internal/dag/gatewayapi_processor.go

```diff
@@ -168,6 +188,12 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 				// with the Gateway. If this Selector is defined, only routes matching the Selector
 				// are associated with the Gateway. An empty Selector matches all routes.
 
+				// Validate the listener protocol is type=TLS.
+				if listener.Protocol != gatewayapi_v1alpha1.TLSProtocolType {
+					p.Errorf("invalid listener protocol %q for Kind: TLSRoute", listener.Protocol)
+					continue
+				}
+
```

Comment:
I'd check this before entering the loop over `p.source.tlsroutes` since it doesn't require any info from each individual route and doing it once per route will result in repeated errors.

Reply:
👍